src/core/agent/services/credentialService.ts

```diff
@@ -179,13 +164,17 @@ class CredentialService extends AgentService {
     if (unSyncedData.length) {
       //sync the storage with the signify data
       for (const credential of unSyncedData) {
-        await this.saveAcdcMetadataRecord(
-          credential.sad.d,
-          credential.sad.a.dt,
-          credential.schema.title,
-          credential.sad.i,
-          credential.schema.$id
-        );
+        await this.createMetadata({
+          id: credential.sad.d,
+          isArchived: false,
+          issuanceDate: new Date(credential.sad.a.dt).toISOString(),
+          credentialType: credential.schema.title,
+          status: CredentialStatus.PENDING,
+          connectionId: credential.sad.i,
+          schema: credential.schema.$id,
+          identifierType: credential.identifierType,
+          identifier: credential.identifier,
```

Comment:
Credential items from KERIA do not have an `identifierType` and `identifier` field. The identifier is the `credential.sad.a.i` field (please verify this!).

To get the identifier type you will need to fetch the identifier from our storage first. Which is OK as this sync should only be needed for edge cases and recovery.